enhancements/insights/insights-runtime-extractor.md

```diff
@@ -0,0 +1,652 @@
+
+The Insights operator can also not tell what is the operating system and its version used by the user's workloads. Knowing this data means that BU and product leadership can use this data to prioritize future work and realign investment.
+
+The proof of concept for this has been demonstrated to Red Hat leeadership, who agree that this is important data for us to know in order 
+to help shape the future of our product offerings.
+
+### User Stories
+
+* As a middleware product manager, I can observe the trends of a product version over time to estimate the expected value of any customer requested extended life offering.
+
+* As a product manager in the AI space, I can measure the application platforms/frameworks/languages used in clusters or namespaces running OpenShift AI to understand what users are coding applications in so we could offer up-to-date images with those libraries and think how about how to support those better.
+
+* As a middleware product manager, I can track the number of applications built using Quarkus to understand our return on investment from an OpenShift workload perspective.
+
+* As a RHEL product manager, I can follow trends in the use of other Linux containers to inform my product roadmap.
+
+* As an OpenShift product marketing manager, I want to be able to produce content that shows users use OpenShift as a platform for all types of workloads.
+    
+
+### Goals
+
+The goal of this proposal is to inspect running containers and surface a few key pieces of data to identify the workload that they are running. This data will enrich and complement the information already gathered by the Insights Operator to get a more accurate representation of workloads running on OpenShift.
+
+### Non-Goals
+
+* It is not a goal of this proposal to scan containers for security purposes
+* It is not a goal of this proposal to get runtime information specific to a given customer or application
+* It is not a goal of this proposal to capture personal sensitive information
+
+## Proposal
+
+This enhancement proposal introduces additional data to identify the technology stack of running containers.
+
+The existing `workloadContainerShape` type that contains workload information about a container will be enhanced with a new `workloadRuntimeInfoContainer` field:
+
+```go
+// workloadContainerShape describes the shape of a container which includes
+// a subset of the data in the container.
+// TODO: this may desirable to make more precise with a whole container hash
+//
+//  that includes more of the workload, but that would only be necessary if
+//  it assisted reconstruction of type of workloads.
+type workloadContainerShape struct {
+   // ImageID is the content addressable hash of the image as observed from
+   // the status or the spec tag.
+   ImageID string `json:"imageID"`
+   // FirstCommand is a hash of the first value in the command array, if
+   // any was set.
+   FirstCommand string `json:"firstCommand,omitempty"`
```

Comment:
nit: This could be called just `Command` ?

Reply:
It is an existing field in this struct.
This RFE only adds the new `workloadRuntimeInfoContainer` field.